core/blockchain.go

```diff
@@ -428,7 +428,20 @@ func NewBlockChain(
 	// Start tx indexer/unindexer if required.
 	if bc.cacheConfig.TxLookupLimit != 0 {
 		bc.wg.Add(1)
-		go bc.maintainTxIndex()
+		go func() {
+			defer bc.wg.Done()
+			var (
+				headCh = make(chan ChainEvent, 1) // Buffered to avoid locking up the event feed
+				sub    = bc.SubscribeChainAcceptedEvent(headCh)
+			)
```

Comment:
```suggestion
		var (
				headCh = make(chan ChainEvent, 1) // Buffered to avoid locking up the event feed
				sub    = bc.SubscribeChainAcceptedEvent(headCh)
		)
		go func() {
			defer bc.wg.Done()
```